paasta_tools/cli/cmds/local_run.py

```diff
@@ -688,12 +694,34 @@ def check_if_port_free(port):
     return True
 
 
+def resolve_aws_account_from_runtimeenv(aws_account: str = None) -> str:
+    if aws_account is not None:
+        return aws_account
+
+    try:
```

Comment:
If we want to do something closer to what @danielpops suggested in the original PR to also assume some right defaults if they also specify a `cluster`, you could use https://github.com/Yelp/paasta/blob/master/paasta_tools/utils.py#L2842-L2843 e.g. `system_paasta_config.get_kube_clusters().get(cluster, {}).get("aws_account_name", local_runtimeenv)` to grab the aws account associated with a particular cluster (stored in our /etc/paasta/kube_clusters.json file), otherwise fall back to the runtimeenv.

Reply:
For maximum compatibility, the AWS account is determined 
1. First by argument
2. If argument is omitted, then get from kube cluster mapping
3. Else get from runtime environment

Reply:
So new behaviour uses the AWS account of the kube cluster, unless it's overridden.